src/FilesystemRepository.php

```diff
@@ -124,10 +119,10 @@ public static function isSymlinkSupported()
      */
     public function __construct($baseDir = '/', $symlink = true, $relative = true)
     {
-        Assert::directory($baseDir);
+        parent::__construct($baseDir);
+
         Assert::boolean($symlink);
 
-        $this->baseDir = rtrim(Path::canonicalize($baseDir), '/');
```

Comment:
I think it make sense to have the same restriction. IMO, the problem is more should we use the same directory for restriction and base directory or should we have another option? Using the same one could be conflicting some times (for instance, the current tests wouldn't work).

What do you think?